packages/core/http/core-http-browser-internal/src/fetch.ts

```diff
@@ -90,6 +90,11 @@ export class Fetch {
           controller
         );
         const initialResponse = this.fetchResponse(interceptedOptions);
+
+        if (interceptedOptions.rawResponse === false) {
+          resolve((await initialResponse) as HttpResponse<TResponseBody>);
+        }
```

Comment:
Yeah, this is what I was mentioning in our slack discussion: you're exiting very early here, bypassing the interceptors calls a few lines below. 

So we're bypassing a few things that we shouldn't. I'm mostly thinking of:

1. security interceptor

https://github.com/elastic/kibana/blob/0a0ead25fdd1bdcada1bd3916dd36492a30b1a2a/x-pack/plugins/security/public/plugin.tsx#L177

2. session timeout 

https://github.com/elastic/kibana/blob/0dfa6374ba4211f23ad2689555bdae495010dad7/x-pack/plugins/security/public/session/session_timeout.ts#L206

3. licensing check

https://github.com/elastic/kibana/blob/8539a912b52a02b372bd4d4756f260948d9ea4ed/x-pack/plugins/licensing/public/plugin.ts#L104-L108

And any other registered http interceptors.

Now, given the usage you're doing of this new option/feature, it's probably fine, but a more globally adoption could introduces edge cases (e.g what happens if we don't detect a session/license change because all requests in a given time interval are "raw"?)

Adding @rudolf for a second opinion here.

(also if we go with this, we gonna need the associated test coverage, as least unit tests for `fetch.test.ts`)

Reply:
Just briefly looked at this but my understanding is the problem we need to solve is that we don't want to wait for and parse the whole response body. Even if interceptors can read the body, I'm not sure anyone uses this and it's an optional property. But having them run on information in the response code or headers is still valuable.

Would it be possible to just skip the body parsing based on the raw flag inside https://github.com/elastic/kibana/blob/main/packages/core/http/core-http-browser-internal/src/fetch.ts#L171-L185

Maybe it even makes sense to remove the response body from the interceptors API (though that could break third-party plugins)?
```
/**
   * Define an interceptor to be executed after a response is received.
   * @param httpResponse {@link HttpResponse}
   * @param controller {@link IHttpInterceptController}
   */
  response?(
    httpResponse: Omit<HttpResponse, 'body'>,
    controller: IHttpInterceptController
  ): MaybePromise<IHttpResponseInterceptorOverrides> | void;
```

Reply:
Addressed in [`4b95cf2` (#158678)](https://github.com/elastic/kibana/pull/158678/commits/4b95cf2cd57f98f7b7f12cd0f62ced001420dd53)

Reply:
I suspect you still need something like
```diff
+++ b/packages/core/http/core-http-browser-internal/src/fetch.ts
@@ -168,7 +168,9 @@ export class Fetch {
     const contentType = response.headers.get('Content-Type') || '';
 
     try {
-      if (NDJSON_CONTENT.test(contentType) || ZIP_CONTENT.test(contentType)) {
+      if (fetchOptions.rawResponse) {
+        body = null; // don't wait for or parse the body
+      } else if (NDJSON_CONTENT.test(contentType) || ZIP_CONTENT.test(contentType)) {
         body = await response.blob();
       } else if (JSON_CONTENT.test(contentType)) {
         body = await response.json();
```

Reply:
Thanks, addressed in d8328735eb9a33309a769144d6d7243d7f33a4a7! I'll punt on making changes to the interceptors API in this PR unless you have strong objections :)

Reply:
Yeah, as long as this last commit did not break any FTR tests, it looks fine to me. LGTM-ing!